src/frontend/src/app/problem/problem.component.ts

```diff
@@ -1,37 +1,89 @@
-import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { LobbyService } from '../services/lobby/lobby.service';
 import { ProblemsService } from '../services/problems/problems.service';
+import { EndData } from '../types/EndData';
+import { PlayerData } from '../types/PlayerData';
+import {
+  MatDialog,
+  MAT_DIALOG_DATA,
+  MatDialogRef,
+} from '@angular/material/dialog';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-problem',
   templateUrl: './problem.component.html',
   styleUrls: ['./problem.component.scss'],
 })
-export class ProblemComponent implements OnInit {
+export class ProblemComponent implements OnInit, OnDestroy {
   title: string = '';
   description: string = '';
   id: string = '';
 
+  players: PlayerData[] = [];
+  endDataSub: Subscription | undefined;
+
   constructor(
     private problemService: ProblemsService,
     private aRoute: ActivatedRoute,
-    private router: Router
+    private router: Router,
+    private lobbyService: LobbyService,
+    public dialog: MatDialog
   ) {}
 
   ngOnInit() {
-    const id = this.aRoute.snapshot.paramMap.get('id');
-    if (id != null) {
-      this.problemService.getProblems().subscribe((problems) => {
-        for (var i = 0; i < problems.length; i++) {
-          if (problems[i]._id === id) {
-            this.title = problems[i].name;
-            this.description = problems[i].description;
-            this.id = problems[i]._id;
+    this.lobbyService.waitingRoom.subscribe((players) => {
+      this.players = players;
+    });
+
+    this.endDataSub = this.lobbyService.endData.subscribe((endData) => {
+      if (endData) {
+        this.dialog.open(EndGameDialogComponent, {
+          data: endData,
+          disableClose: true,
+        });
+      }
+    });
+
+    this.aRoute.paramMap.subscribe((params) => {
+      const id = params.get('id');
+      if (id != null) {
+        //Close dialogs
+        this.dialog.closeAll();
+        this.problemService.getProblems().subscribe((problems) => {
+          for (var i = 0; i < problems.length; i++) {
+            if (problems[i]._id === id) {
```

Comment:
Shouldnt be getting all problems and then filtering on the frontend. Get by id from the backend/db directly. 

Seems like this was here before this diff but we should fix this at some point.